src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+#include "Quadrature.h"
+
+
+/**
+ * @brief Dummy constructor sets the default number of angles to zero.
+ */
+Quadrature::Quadrature() {
+  _num_azim = 0;
+  _num_polar = 0;
+  _sin_thetas = NULL;
+  _thetas = NULL;
+  _phis = NULL;
+  _azim_spacings = NULL;
+  _azim_weights = NULL;
+  _polar_weights = NULL;
+  _total_weights = NULL;
+}
+
+
+/**
+ * @brief Destructor deletes arrray of sines of the polar angles, the weights
+ *        of the polar angles and the products of the sines and weights.
+ */
+Quadrature::~Quadrature() {
+
+  if (_sin_thetas != NULL) {
+    for (int a=0; a < _num_azim/2; a++)
+      delete [] _sin_thetas[a];
+    delete [] _sin_thetas;
+  }
+
+  if (_thetas != NULL) {
+    for (int a=0; a < _num_azim/2; a++)
+      delete [] _thetas[a];
+    delete [] _thetas;
+  }
+
+  if (_phis != NULL)
+    delete [] _phis;
+
+  if (_azim_spacings != NULL)
+    delete [] _azim_spacings;
+
+  if (_azim_weights != NULL)
+    delete [] _azim_weights;
+
+  if (_polar_weights != NULL) {
+    for (int a=0; a < _num_azim/2; a++)
+      delete [] _polar_weights[a];
+    delete [] _polar_weights;
+  }
+
+  if (_total_weights != NULL) {
+    for (int a=0; a < _num_azim/2; a++)
+      delete [] _total_weights[a];
+    delete [] _total_weights;
+  }
+}
+
+
+
+/**
+ * @brief Returns the number of polar angles.
+ * @return the number of polar angles
+ */
+int Quadrature::getNumPolarAngles() const {
+  return _num_polar;
+}
+
+
+/**
+ * @brief Returns the number of azimuthal angles.
+ * @return the number of azimuthal angles
+ */
+int Quadrature::getNumAzimAngles() const {
+  return _num_azim;
+}
+
+
+/**
+ * @brief Returns the \f$ sin(\theta)\f$ value for a particular polar angle.
+ * @param azim index of the azimthal angle of interest
+ * @param polar index of the polar angle of interest
+ * @return the value of \f$ \sin(\theta) \f$ for this azimuthal and polar angle
+ */
+FP_PRECISION Quadrature::getSinTheta(int azim, int polar) {
+
+  if (polar < 0 || polar >= _num_polar)
+    log_printf(ERROR, "Attempted to retrieve sin theta for polar angle = "
+               "%d and azim angle = %d when only %d polar angles are "
+               "defined", polar, azim, _num_polar);
+
+  else if (azim < 0 || azim >= _num_azim)
+    log_printf(ERROR, "Attempted to retrieve sin theta for polar angle = "
+               "%d and azim angle = %d when only %d azim angles are "
+               "defined", polar, azim, _num_azim);
+
+  else if (_sin_thetas == NULL)
+    log_printf(ERROR, "Attempted to retrieve sin theta for polar angle = %d "
+               "and azim angle = %d but the sin thetas have not been "
+               "initialized", polar, azim);
+
+  return _sin_thetas[azim][polar];
+}
+
+
+/**
+ * @brief Returns the polar angle value for a given azimuthal and polar
+ *        angle index.
```

Comment:
Add to this comment whether the angle is in units of radians or degrees (radians I believe).